swap/src/api/request.rs

```diff
@@ -0,0 +1,861 @@
+                    .await
+                    .expect("Could not release swap lock");
+
+                state.map(|state| {
+                    json!({
+                        "result": state,
+                    })
+                })
+            }
+            Method::History => {
+                let swaps = context.db.all().await?;
+                let mut vec: Vec<(Uuid, String)> = Vec::new();
+                for (swap_id, state) in swaps {
+                    let state: BobState = state.try_into()?;
+                    vec.push((swap_id, state.to_string()));
+                }
+
+                Ok(json!({ "swaps": vec }))
+            }
+            Method::GetRawStates => {
+                let raw_history = context.db.raw_all().await?;
+
+                Ok(json!({ "raw_states": raw_history }))
+            }
+            Method::Config => {
+                let data_dir_display = context.config.data_dir.display();
+                tracing::info!(path=%data_dir_display, "Data directory");
+                tracing::info!(path=%format!("{}/logs", data_dir_display), "Log files directory");
+                tracing::info!(path=%format!("{}/sqlite", data_dir_display), "Sqlite file location");
+                tracing::info!(path=%format!("{}/seed.pem", data_dir_display), "Seed file location");
+                tracing::info!(path=%format!("{}/monero", data_dir_display), "Monero-wallet-rpc directory");
+                tracing::info!(path=%format!("{}/wallet", data_dir_display), "Internal bitcoin wallet directory");
+
+                Ok(json!({
+                    "log_files": format!("{}/logs", data_dir_display),
+                    "sqlite": format!("{}/sqlite", data_dir_display),
+                    "seed": format!("{}/seed.pem", data_dir_display),
+                    "monero-wallet-rpc": format!("{}/monero", data_dir_display),
+                    "bitcoin_wallet": format!("{}/wallet", data_dir_display),
+                }))
+            }
+            Method::WithdrawBtc { address, amount } => {
+                let bitcoin_wallet = context
+                    .bitcoin_wallet
+                    .as_ref()
+                    .context("Could not get Bitcoin wallet")?;
+
+                let amount = match amount {
+                    Some(amount) => amount,
+                    None => {
+                        bitcoin_wallet
+                            .max_giveable(address.script_pubkey().len())
+                            .await?
+                    }
+                };
+                let psbt = bitcoin_wallet
+                    .send_to_address(address, amount, None)
+                    .await?;
+                let signed_tx = bitcoin_wallet.sign_and_finalize(psbt).await?;
+
+                bitcoin_wallet
+                    .broadcast(signed_tx.clone(), "withdraw")
+                    .await?;
+
+                Ok(json!({
+                    "signed_tx": signed_tx,
```

Comment:
I don't think we need to return the whole tx here. Amount and txid should be enough. Returning the fee as well could be useful